### DISCUSSION TITLE
Menu - Show/Hide menu on hover issue

### DISCUSSION THREAD

**Discussion by binhxn** [ariakit/ariakit#2230] [Questions] [answered]
Hello 👋

I'm attempting to use Menu to show/hide on hover but running into a hover issue. I'd like to:
1. When hovering over the MenuButton, the Menu shows
2. While Menu is open and hovering between both Menu and MenuButton, the Menu is still open
3. While Menu is open, hovering outside the Menu will close

I'm stuck on the second point because there's a small window where the menu toggles between the show/hide states. 

https://user-images.githubusercontent.com/8155387/228626925-e07b4616-405b-4c02-9933-6b784970c08a.mp4

I tried using a combination of  `showOnHover` on MenuButton and `menu.hide()` on Menu, but that didn't work. I also tried using `hideOnHoverOutside` on Menu, but that didn’t seem to do anything (looks like it only works with HoverCard). Here's a [codesandbox](https://codesandbox.io/s/muddy-sound-wxqc12?file=/src/App.js) to reproduce the issue. Would love to get your feedback!

**Comment by diegohaz** [comment, marked as answer]
Hi @binhxn!

Here are some suggestions:

- `modal` doesn't make sense when a popup is open on hover. If you want a portalled menu, use the `portal` prop. If you really want a `modal` menu and you're building a hybrid menu (that also works on click), I recommend using it only when the menu is **not** triggered by hover:

  ```jsx
  const [hovered, setHovered] = useState(false);
  
  <div>
    <MenuButton
      showOnHover={() => {
        setHovered(true);
        return true;
      }}
      onClick={() => setHovered(false)}
    />
    <Menu modal={!hovered} />
  </div>
  ```

- I recommend disabling `focusOnHover` on the menu items. It's annoying when the focus changes when you're just hovering over a page. Again, if you're building a hybrid menu, you can do something like `focusOnHover={!hovered}`. Or, better, enable `focusOnHover` only when menu already has focus:

  ```jsx
  <MenuItem focusOnHover={() => menuRef.current?.contains(document.activeElement)} />
  ```

- `hideOnHoverOutside` has no effect if the hovercard has focus. That's by design. If you follow the suggestions above, there will be no unintended focus on hover, so this prop will work.

- You can play with `useMenuState({ showTimeout, hideTimeout })` if you want.

CodeSandbox with a minimal solution: https://codesandbox.io/s/ariakit-menu-showonhover-ye0xrz?file=/src/App.js

**Reply by binhxn** [reply]
The Codesandbox solution you provided is exactly what I'm looking for! I plan on creating a hybrid menu at some point, so these suggestions are super helpful. Thank you for your help!

**Reply by diegohaz** [reply]
@binhxn Nice! Would you mind sharing more details about your specific use case? We can probably use it as an inspiration to create an example for the site.

**Reply by binhxn** [reply]
@diegohaz Sure! At the moment, I have a requirement to create a responsive navbar where:
1. When the browser is **_above_** a breakpoint, hovering over a navbar item will open the menu
2. When the browser is **_below_** a certain breakpoint, some navbar item collapses. Only by _**clicking**_ the "More Options" icon, will it display the collapsed navbar items through menus and submenus

The video below shows an example of this:

https://user-images.githubusercontent.com/8155387/228960991-85970f3f-fc78-4eaa-882d-a61f63920655.mp4


It wasn't recorded, but I also want to point out that keyboard navigation continues to work as expected 👍
